### PR TITLE
- Fix RabbitMQ server version check logic

### PR DESCRIPTION
**Pull request by jas88**
- Avoid ignoring exceptions we should abort on
- Simplifying, trying to plug the thread leaks

## Proposed Changes

Summarise your proposed changes here, including any notes for reviewers.

## Types of changes

What types of changes does your code introduce? Tick all that apply.

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation-Only Update (if none of the other choices apply)
  - In this case, ensure that the message of the head commit from the source branch is prefixed with `[skip ci]`

## Checklist

By opening this PR, I confirm that I have:

- [x] Reviewed the [contributing](https://github.com/SMI/SmiServices/blob/master/CONTRIBUTING.md) guidelines for this repository
- [X] Ensured that the PR branch is in sync with the target branch (i.e. it is automatically merge-able)
- [X] Updated any relevant API documentation
- [X] Created or updated any tests if relevant
- [x] Created a [news](https://github.com/SMI/SmiServices/blob/master/news/README.md) file
    -   NOTE: This ***must*** include any changes to any of the following files: default.yaml, any of the RabbitMQ server configurations, GlobalOptions.cs
- [X] Listed myself in the [CONTRIBUTORS](https://github.com/SMI/SmiServices/blob/master/CONTRIBUTORS.md) file 🚀
- [X] Requested a review by one of the repository maintainers

## Issues

If relevant, tag any issues that are *expected* to be resolved with this PR. E.g.:

- Closes #\<issue-number>
- ...

### CODE REVIEW

File: src/common/com.smi.microservices.common/src/main/java/org/smi/common/execution/SmiShutdownHook.java [hunk 1]
```diff
@@ -16,8 +18,11 @@ public SmiShutdownHook(IMicroserviceHost host) {
 	 */
 	@Override
 	public void run() {
-
 		if (_host != null)
-			_host.Shutdown();
+			try {
+				_host.Shutdown();
+			} catch (IOException e) {
+				e.printStackTrace();
```
Comment (rkm) [inline src/common/com.smi.microservices.common/src/main/java/org/smi/common/execution/SmiShutdownHook.java:25]: Does this get written to the log file? I don't think we always capture the stderr at the moment so this info may be lost.

File: src/common/com.smi.microservices.common/src/main/java/org/smi/common/rabbitMq/RabbitMqAdapter.java [hunk 2]
```diff
@@ -76,32 +66,23 @@ public RabbitMqAdapter(RabbitOptions options, String microserviceName) throws IO
...
+		final int MinRabbitServerVersionMinor = 7;
+		final int MinRabbitServerVersionPatch = 0;
+
+		if (!_conn.getServerProperties().containsKey("version"))
+			throw new IOException("Could not get RabbitMQ server version");
+
+		_rabbitMqServerVersion = _conn.getServerProperties().get("version").toString();
+		String[] split = _rabbitMqServerVersion.split(Pattern.quote("."));
+
+		if (Integer.parseInt(split[0]) < MinRabbitServerVersionMajor
```
Comment (rkm) [inline src/common/com.smi.microservices.common/src/main/java/org/smi/common/rabbitMq/RabbitMqAdapter.java:80]: There's similar logic in the c# code which we should also fix if it's broken.

**Review by rkm — CHANGES_REQUESTED** [review]

File: src/common/com.smi.microservices.common/src/main/java/org/smi/common/rabbitMq/RabbitMqAdapter.java (hunk 2 above)
Comment (jas88) [reply src/common/com.smi.microservices.common/src/main/java/org/smi/common/rabbitMq/RabbitMqAdapter.java:80]: Good point - fixed in this PR now.

File: src/common/com.smi.microservices.common/src/main/java/org/smi/common/execution/SmiShutdownHook.java (hunk 1 above)
Comment (jas88) [reply src/common/com.smi.microservices.common/src/main/java/org/smi/common/execution/SmiShutdownHook.java:25]: All we're doing at that point is closing the Rabbit connection - if it fails I don't think we have much to log anyway. (It wasn't logged earlier, either, just printed to stderr - I've just moved the call "up" a bit in the call stack.)

**Comment by rkm** [comment]
@jas can you rebase this to bump the CI checks?

File: news/1023-change.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+Fix RabbitMQ version check logic
+Tidy Java code
```
Comment (rkm) [inline news/1023-change.md:2]: just a note for future that the github markdown parser considers these to be one line, so when the changelog is generated they will be collapsed together

Comment (jas88) [reply news/1023-change.md]: Good point, thanks - I've switched it to a slightly more detailed sublist which might make more sense anyway. OK to merge now?

Comment (rkm) [reply news/1023-change.md]: good to go 👍

**Review by rkm — APPROVED** [review]

**Pull request merged**